front/src/pages/EditItem/EditItems.tsx

```diff
@@ -0,0 +1,158 @@
+// file was renamed from "EditItem.tsx" to "EditItems.tsx"
+// cause: typescript didn't work if we had the previous name
+
+import React, { useEffect, useState } from 'react';
+import { Link, useNavigate, useParams } from 'react-router-dom';
+import styles from './EditItem.module.scss';
+import useFetchTodos from '../../hooks/useFetchTodos';
+import { importanceValues } from '../../constants';
+import { todosAPI } from '../../api/todos-api';
+
+const EditItem = () => {
+  const { todoId } = useParams();
+  const {
+    data: todos = []
+  } = useFetchTodos();
+  const currentTodo = todos.find(todo => todo.id === todoId);
+  const [todoState, setTodoState] = useState({
+    title: currentTodo?.title || '',
+    importance: currentTodo?.importance || '',
+    isDone: currentTodo?.isDone || false,
+  })
+  const navigate = useNavigate();
+
+  let pageTitle: string;
+  let pageContent: JSX.Element;
+  if (currentTodo) {
+    pageTitle = 'Edit your todo';
+    pageContent = getEditForm();
+  } else {
+    pageTitle = 'Chose todo to edit';
+    pageContent = getTodosList();
+  }
+
+  useEffect(() => {
+    setTodoState({
+      title: currentTodo?.title || '',
+      importance: currentTodo?.importance || '',
+      isDone: currentTodo?.isDone || false,
+    });
+  }, [todoId]);
+
+  function handleFormSubmit (e: React.FormEvent) {
+    e.preventDefault();
+    const changedTodo = {
+      ...todoState,
+      id: todoId as string,
+    }
+    todosAPI.updateTodo(changedTodo)
+      .then(() => {
+        navigate('/');
+      })
+  }
+
+  function handleInputChange (e: React.ChangeEvent<HTMLInputElement>) {
```

Comment:
I think having separate `handleTitleChange` and `handleDoneStateChange` is better. Generic function targets DRY, but has worse flexibility(you already have here one additional `if` and with more complex models it would be worse). Also because of generic name, readability is also under attack :)